### DISCUSSION TITLE
Load asset entity position/bounding box

### DISCUSSION THREAD

**Discussion by amenon** [google/filament#4654] [Q&A] [answered]
I've loaded a GLB file into a TextureView on Android. I am looking for help on determining the bounding box of a single entity in the model.

The `FilamentAsset` API allows me to get the entity ids and their names, but I seem to only be able to get the bounding box of the entire asset instead of more granular position of a single entity.

For context, I have multiple objects in the scene that have known entity names, and I am looking to place the camera in an ideal position to focus on the main subject to minimize occlusions. 

Does anyone have any suggestions to extract bounding box information for an entity?

**Comment by pixelflinger** [comment, marked as answer]
You should able to use `getAxisAlignedBoundingBox` on `RenderableManager`?

**Reply by amenon** [reply]
Thank you for that suggestion! I want to make sure that I am attempting this the correct way. I modified the `populateScene` method in `ModelViewer` to add `getAxisAlignedBoundingBox`.

```
    private fun populateScene(asset: FilamentAsset) {
        val rcm = engine.renderableManager
        var count = 0
        val popRenderables = { count = asset.popRenderables(readyRenderables); count != 0 }
        while (popRenderables()) {
            for (i in 0 until count) {
                val entity = readyRenderables[i]
                val ri = rcm.getInstance(entity)
                rcm.setScreenSpaceContactShadows(ri, true)

                val nodeName = asset.getName(entity).lowercase()
                val box = rcm.getAxisAlignedBoundingBox(ri, null)
                Timber.d(
                    "%s Center: %s; Half: %s",
                    nodeName,
                    box.center.joinToString(),
                    box.halfExtent.joinToString()
                )
            }
            scene.addEntities(readyRenderables.take(count).toIntArray())
        }
        scene.addEntities(asset.lightEntities)
    }
```

Here is an example result from one of my GLB assets:

> model:female_mesh Center: 0.0, 0.7952248, -0.039767593; Half: 0.69250405, 0.7950821, 0.1899653
> wall Center: 3.4332275E-5, 1.225, 0.0; Half: 3.2176454, 1.225, 0.038648985
> floor Center: -1.3969839E-9, 0.01, -1.8626451E-9; Half: 0.010000002, 0.0, 0.010000003

In this case, the `model` node is behind the wall. However, when I attempt to do this for another GLB file that has the `model` node in front of the wall, I get the exact same `center`s and `halfExtent`s. 

So I was wondering if I am getting these before they are actually placed in the scene?

**Reply by romainguy** [reply]
The bounding boxes you get this way are local to the object. They are not in world space. To get that you need to transform the bounding box by the world transform of the renderable that you can obtain via `TransformManager`.

**Reply by amenon** [reply]
Thanks @romainguy and @pixelflinger . That was very helpful. I was able to use the `TransformManager` to get the relative positioning that I needed.

**Reply by MarxMc** [reply]
Is bounding box displayed over the 3d Object ?
